tests/rptest/tests/scram_test.py

```diff
@@ -496,3 +498,47 @@ def test_invalid_password(self):
             password=password,
             expected_status_code=400,
             err_msg='Parameter contained invalid control characters: PASSWORD')
+
+
+class SCRAMReauthTest(BaseScramTest):
+    EXAMPLE_TOPIC = 'foo'
+
+    def __init__(self, test_context, **kwargs):
+        security = SecurityConfig()
+        security.enable_sasl = True
+        super().__init__(test_context=test_context,
+                         num_brokers=3,
+                         security=security,
+                         extra_rp_conf={'kafka_sasl_max_reauth_ms': 8000},
+                         **kwargs)
+
+        username, password, algorithm = self.redpanda.SUPERUSER_CREDENTIALS
+        self.rpk = RpkTool(self.redpanda,
+                           username=username,
+                           password=password,
+                           sasl_mechanism=algorithm)
+
+    @cluster(num_nodes=3)
+    def test_scram_reauth(self):
+        self.rpk.create_topic(self.EXAMPLE_TOPIC)
+        su_client = self.make_superuser_client()
+        producer = su_client.get_producer()
+        producer.poll(0.0)
+
+        metrics = get_sasl_metrics(self.redpanda)
+
+        for i in range(0, 200):
+            producer.poll(1.0)
+            producer.produce(topic=self.EXAMPLE_TOPIC, key='bar', value=str(i))
+            time.sleep(0.1)
+
+        producer.flush(timeout=2)
+
+        metrics = get_sasl_metrics(self.redpanda)
+        self.redpanda.logger.debug(f"SASL metrics: {metrics}")
+        assert (EXPIRATION_METRIC in metrics.keys())
+        assert (metrics[EXPIRATION_METRIC] == 0
+                ), "Client should reauth before session expiry"
+        assert (REAUTH_METRIC in metrics.keys())
+        assert (metrics[REAUTH_METRIC]
+                > 0), "Expected client reauth on some broker..."
```

Comment:
I'm surprised the client is reauthenticating. 

Do we have to modify `scram_authenticator` or does the client not send an expiry?

Reply:
AFAIK the client doesn't send an expiry with SCRAM creds, but it still receives back a session lifetime based on the cluster config.